src/Solr/Reindex/Handlers/SolrReindexBase.php

```diff
@@ -236,6 +237,17 @@ protected function getRecordsInGroup(SolrIndex $indexInstance, $class, $groups,
             $items = $items->filter('ClassName', $class);
         }
 
+        // ShowInSearch filter
+        // we cannot use $items->remove($item), as that deletes the record from the database
+        $ids = $items->sort('ID')->column('ID');
+        $ids = array_fill_keys($ids, true);
+        foreach ($items as $item) {
+            if (isset($item->ShowInSearch) && !$item->ShowInSearch) {
```

Comment:
Same concern here, that likely ignores any overloaded getters?

Reply:
overloaded getters still work, see https://github.com/silverstripe/silverstripe-fulltextsearch/issues/257#issuecomment-601488939